aten/src/ATen/native/mps/operations/RangeFactories.mm

```diff
@@ -13,6 +13,42 @@
 namespace at {
 namespace native {
 
+static const char* ARANGE_KERNEL = R"METAL(
+#include <metal_stdlib>
+
+using namespace metal;
+struct ARangeFloatParam {
+  float start;
+  float step;
+  uint length;
+};
+
+kernel void arange_float(constant ARangeFloatParam *params [[buffer(0)]],
+                         device float  *out [[buffer(1)]],
+                         uint offset[[thread_position_in_grid]]) {
+  if (offset >= params->length) {
+    return;
+  }
+  out[offset] = params->start + offset * params->step;
+}
+)METAL";
+
+static id<MTLComputePipelineState> compileMetalShader(id<MTLDevice> device) {
+ static id<MTLComputePipelineState> rc = nil;
+ if (rc != nil) {
+    return rc;
+ }
+ NSError *error = nil;
+ id<MTLLibrary> library = [device newLibraryWithSource:[NSString stringWithUTF8String:ARANGE_KERNEL]
```

Comment:
I think when we load the MPS device, we can pre-warm the library with offline compilation of the Metal kernels and load the functions here.